### ISSUE TITLE
Unable to insert record on Xcode 11.2 beta 2 build 11B44 targeting iOS 13.2

### ISSUE THREAD

**Issue by mhrakoto999** [groue/GRDB.swift#636]
### What did you do?

It’s a regular insertion 
let library = Library()
try library.insert(db)

### What did you expect to happen?

I expected the insertion to succeed

### What happened instead?

The error happens when calling try DAO(db, self) within encode(to: encoder) called by encode(to container:à in the extension EncodableRecord.

The program throws either EXC_BAD_ACCESS (code: EXC_1386_GPFLT) or EXC_BAD_ACCESS (code=257 address: 0x1044e4ca9)

### Environment

**GRDB flavor(s):** (GRDB, SQLCipher, Custom SQLite build?)
GRDB

**GRDB version:**
4.4.0

**Installation method:** (CocoaPods, SPM, manual?)
SPM

**Xcode version:**
11.2 beta 2 (11B44)

**Swift version:**
5.1

**Platform(s) running GRDB:** (iOS, macOS, watchOS?)
iOS 

**macOS version running Xcode:**
10.15 beta (19A582a)

### Demo Project

My code for the entity 

struct Library: Codable, Identifiable {
    var id: UUID
    
    var createdAt: Date
    var lastModified: Date
    
    init() {
        self.id = UUID()
        self.createdAt = Date()
        self.lastModified = self.createdAt
    }
}

Columns definition in the database:

try db.create(table: "library")
{ t in
                t.column("id", .text)
                    .primaryKey()
                    .notNull()
                
                t.column("createdAt" , .datetime)
                    .notNull()
                
                t.column("lastModified", .datetime)
                    .notNull()
            }

And these additional conformances:

//MARK: - RELATIONSHIPS
extension Library: TableRecord {
    
    /// To-many relationship to `Book`
    static let books = hasMany(Book.self)
    
    var books: QueryInterfaceRequest<Book> { request(for: Library.books) }
    
}

//MARK: - DATABASE OPERATIONS
// Allows database support for free as the struct conforms already to Codable
extension Library: FetchableRecord { }

extension Library: PersistableRecord {
    
    // encodes UUID to string as database value
    static let databaseUUIDEncodingStrategy: DatabaseUUIDEncodingStrategy = .string
    
}

**Comment by groue** [comment]
Hello @mhrakoto999,

You are running beta software, so it's unclear where the bug is. And Apple bugs are not fixed here, as you certainly know. If you are looking for support, I suggest you come back with:

1. some self-contained sample code that reproduces the crash.
2. a proper crash log with stack trace and eventual error message.
3. the reason why you think the crash is caused by GRDB.

Thank you

**Comment by groue** [comment]
Also, please wrap your code snippet into backticks, so that Github provides proper rendering and syntax highlighting.

For example, this:

    ```swift
    // Allows database support for free as the struct conforms already to Codable
    extension Library: FetchableRecord { }
    ```

is rendered as:

```swift
// Allows database support for free as the struct conforms already to Codable
extension Library: FetchableRecord { }
```

Much better, isn't it? This will tremendously improve your Github messages.

**Comment by mhrakoto999** [comment]
Hello Gwendal,

I hope this will help

[badAccessGRDB.xcodeproj.zip](https://github.com/groue/GRDB.swift/files/3728130/badAccessGRDB.xcodeproj.zip)

**Comment by groue** [comment]
@mhrakoto999, source files are missing from your zip file. And I asked for **three** things, in case you did not notice.

**Comment by mhrakoto999** [comment]
Oh, I'm sorry

1. Here is a complete zip with source files
[badAccessGRDB.zip](https://github.com/groue/GRDB.swift/files/3728271/badAccessGRDB.zip)

2. Here is a crashlog when I ran the extended program on the simulator 
[badAccessGRDB.pdf](https://github.com/groue/GRDB.swift/files/3728239/badAccessGRDB.pdf)

3. the reason why you think the crash is caused by GRDB ?
a step by step debugging shows that the crash happened when the encoder at the specific GRDB's EncodableRecord protocol is called.

**Comment by groue** [comment]
@mhrakoto999, the zip file does not contain any project. Please send something **self-contained**, will you? Also, you did not say how to use the sample app in order to produce the crash.

The idea, if you want help, is that you help me as well. That would just be fair.

> a step by step debugging shows that the crash happened when the encoder at the specific GRDB's EncodableRecord protocol is called.

Oh, so you did not do any investigation on your own. Remember: you run **beta** software. Maybe you have found a bug somewhere else, which triggers a crash in valid GRDB code. I don't know. If you are *able* do some investigation, that will be appreciated. Otherwise, you may prefer going back to non-beta versions of your favorite tools, the only ones that are not supposed to contain any bug.

**Comment by mhrakoto999** [comment]
OK I'm new here

You just run the sample app : it creates the database with one table 'library' (which has a uuid column and 2 dates) and then populates this table. All these are done via the migrator DatabaseMigrator provided by GRDB. The app crashes when attempting to insert the first record via a regular "try library.save(db)". Therefore the app crashes at launch.

Unfortunately, when the debugger shows that the call to the method encoder(to:) in the EncodableRecord protocol throws the EXC_BAD_ACCESS, I ran out of idea and then filed this issue. 
Really sorry to have bothered you, i'll go it alone

**Comment by groue** [comment]
Yes, I gave you a lot of work ;-) I'll try to mix your two zip files, I suppose they both provide a complete sample project.

**Comment by groue** [comment]
@mhrakoto999, I could reproduce the crash in your app and Xcode 11.2 beta 2.

However, in isolation, the Library type can be inserted without any problem (as in the sample code below).

<details>
    <summary>Code</summary>

```swift
struct Library: Codable, Identifiable {
    var id: UUID
    var createdAt: Date
    var lastModified: Date
    
    init() {
        self.id = UUID()
        self.createdAt = Date()
        self.lastModified = self.createdAt
    }
}

extension Library: TableRecord { }
extension Library: FetchableRecord { }
extension Library: PersistableRecord {
    static let databaseUUIDEncodingStrategy: DatabaseUUIDEncodingStrategy = .string
}

let dbQueue = try DatabaseQueue(...)
try dbQueue.write { db in
    try db.create(table: "library") { t in
        t.column("id", .text)
            .primaryKey()
            .notNull()
        t.column("createdAt", .datetime)
            .notNull()
        t.column("lastModified", .datetime)
            .notNull()
    }
    
    let library = Library()
    try library.insert(db) // success
}
```

</details>

Something happens, *in the context of your application*, that creates the crash. I don't know what it is. Further information is needed.

**Comment by groue** [comment]
@mhrakoto999, the demo GRDBCombine application suffers from the same problem. I'll thus perform my own investigation. Your active help is still welcome, of course.

**Comment by groue** [comment]
This is a bug in the compiler, not in GRDB. The fix is not here. It will take some time to reduce the crash to a tiny sample code that I can submit to Apple for their investigation. Meanwhile, don't use Xcode 11.2 beta 2: it just does not work properly.

Thanks for the report, @mhrakoto999.

**Comment by mhrakoto999** [comment]
Hello Gwendal,
Thank you for your answer.
I will look at it more thoroughly this afternoon or this evening. Now, I realize that the method encoder(to:) crashes because it is fed with an empty dictionary as if there is an unexpected type that should be handled by a new encoding method. This is not the case with previous versions (is it Xcode ? Or Swift itself ?)

**Comment by groue** [comment]
The bug has been reported as [SR-11643](https://bugs.swift.org/browse/SR-11643). We can hope that a future Xcode release will fix it.

If you were to remember anything from this, @mhrakoto999, let it be that a **minimal reproducing code** is super useful. Everything that is not strictly necessary should be removed. This takes time, but now people in charge are *clearly identified* (read: it's not the author of GRDB), and they has something very clear and focused to chew on.

And of course, always provide a **super easy and reliable series of reproduction steps located in a single place**. We already have enough things to think about, and chasing pieces of splattered information is not a funny task.

You will see: this will greatly improve your bug reports, and the efficiency of the resolution of your troubles.

Since this issue is not a GRDB issue, and the actual issue was reported in the correct place, I can close it now.

**Comment by gverdouw** [comment]
I think I am running in to this on the released Xcode 11.2, the swift bug [SR-11643](https://bugs.swift.org/browse/SR-11643) hasn't been closed so I am assuming it is the same issue

**Comment by groue** [comment]
Thank you @gverdouw. Follow up is in #640.
